app/controllers/programming_groups_controller.rb

```diff
@@ -65,21 +81,42 @@ def create
     end
   end
 
+  def update
+    myparams = programming_group_params2
+    myparams[:users] =
+      @programming_group.programming_group_memberships.where(id: myparams[:programming_group_membership_ids].compact_blank).includes(:user).map(&:user)
+    myparams.delete(:programming_group_membership_ids)
+    update_and_respond(object: @programming_group, params: myparams)
+  end
+
+  def destroy
+    authorize!
+    destroy_and_respond(object: @programming_group)
+  end
+
   private
 
   def authorize!
-    authorize(@programming_group)
+    authorize(@programming_group || @programming_groups)
   end
 
   def programming_group_params
     params.require(:programming_group).permit(:programming_partner_ids)
   end
 
+  def programming_group_params2
```

Comment:
A method name with a `2` suffix is not that great and recognizable. Please describe what they are intended for, e.g., `programming_group_params_for_update`.